test/val/val_limits_test.cpp

```diff
@@ -76,3 +76,28 @@ TEST_F(ValidateLimits, idEqualToBoundBad) {
       HasSubstr("Result <id> '64' must be less than the ID bound '64'."));
 }
 
+TEST_F(ValidateLimits, structNumMembersGood) {
+  string str = header + R"(
+%1 = OpTypeInt 32 0
+%2 = OpTypeStruct)";
+  for (int i = 0; i < 16383; ++i) {
+    str += " %1";
```

Comment:
This will cause a quadratic number of memory operations.
Instead, use a std::ostringstream to accumulate the values, and then construct the string once with .str()